scrapy/utils/conf.py

```diff
@@ -26,6 +27,14 @@ def arglist_to_dict(arglist):
     return dict(x.split('=', 1) for x in arglist)
 
 
+def json_arglist_to_dict(json_arglist):
+    """Convert a list of arguments like
+    ['arg1=["first", "second"]', 'arg2={"key": "value"}'] to a dict
+    """
+    dict_ = arglist_to_dict(json_arglist)
+    return dict(zip(dict_.iterkeys(), map(json.loads, dict_.itervalues())))
```

Comment:
Is it the same as `{k: json.loads(v) for k,v in dct.items()}`? We can use Python 2.7 syntax because Python 2.6 is no longer supported.